docs/guide/snaps-development-guide.md

```diff
@@ -250,16 +261,20 @@ Via the MetaMask settings page, the user can see their installed snaps. For each
 - see its execution status (running, stopped, or crashed)
 - enable and disable the snap
 
-Other than the settings page, the only way a snap can modify the MetaMask UI is by creating a confirmation using the [`snap_confirm`](./snaps-rpc-api.html#snap-confirm) RPC method.
-This means that many snaps will have to rely on web pages (i.e., dapps) and their own RPC methods to present their data to the user.
+Other than the settings page, there are only a few select ways a snap can modify the MetaMask UI:
+
+- By opening a dialog using the [`snap_dialog`](./snaps-rpc-api.html#snap-dialog) RPC method.
+- By returning transaction insights from the [`onTransaction`](./snaps-exports.md#ontransaction) export.
+
+This means that most snaps will have to rely on web pages (i.e., dapps) and their own RPC methods to present their data to the user.
 
 Providing more ways for snaps to modify the MetaMask UI is an important goal of the snaps system, and over time more and more snaps will be able to contain their user interfaces entirely within MetaMask itself.
 
 #### Detecting the User's MetaMask Version
 
-When developing a website that depends on Snaps, it's important to know whether MetaMask Flask is installed.
+When developing a website/Dapp that depends on Snaps, it's important to know whether MetaMask Flask is installed.
 For this purpose, we recommend using the [`@metamask/detect-provider`](https://npmjs.com/package/@metamask/detect-provider)
-package [`web3_clientVersion`](https://metamask.github.io/api-playground/api-documentation/#web3_clientVersion)
+package's [`web3_clientVersion`](https://metamask.github.io/api-playground/api-documentation/#web3_clientVersion)
 RPC method as demonstrated in the following snippet:
```

Comment:
This makes it sound like the `@metamask/detect-provider` package provides the `web3_clientVersion` RPC method, but that's not true — the functionality is actually coming from the provider via MetaMask. Maybe this should say:

``` suggestion
We recommend calling the [`web3_clientVersion`](https://metamask.github.io/api-playground/api-documentation/#web3_clientVersion) RPC method to obtain this information. Here we are using the [`@metamask/detect-provider`](https://npmjs.com/package/@metamask/detect-provider) package to get the provider object from MetaMask first:
```

Alternatively, at the top of this document, in "The Snap Source Code", there is an example of using `ethereum.request` to make a request. Why isn't `detect-provider` being used there? If we know we are talking to MetaMask, why can't we use `ethereum.request` directly here?